pkg/reconciler/run/run.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package run
+
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	clientset "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	runreconciler "github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1alpha1/run"
+	corev1 "k8s.io/api/core/v1"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/tracker"
+)
+
+const (
+	// runAgentName defines logging agent name for Run Controller
+	runAgentName = "run-controller"
+)
+
+// Reconciler implements controller.Reconciler for Run resources.
+type Reconciler struct {
+	pipelineClientSet clientset.Interface
```

Comment:
`s/pipelineClientSet/runClientSet/` 🤔

Reply:
This is actually named `pipeline` because it's the "Tekton Pipelines" client -- it provides (generated) methods to interact with _all_ "Tekton Pipelines" resources and API versions: https://godoc.org/github.com/tektoncd/pipeline/pkg/client/clientset/versioned

It's also named `pipelineClientSet` in the TaskRun controller too, for example: https://github.com/tektoncd/pipeline/blob/732d2363b9df0b5c2e6b1de732a819b69c70bb3f/pkg/reconciler/taskrun/taskrun.go#L64

Reply:
thanks for the explanation 👍